packages/core/src/fhircast/index.ts

```diff
@@ -131,66 +143,136 @@ export type SubscriptionRequest = {
+      : EventName extends 'ImagingStudy-open'
+        ? FhircastImagingStudyOpenContext
+        : EventName extends 'ImagingStudy-close'
+          ? FhircastImagingStudyCloseContext
+          : EventName extends 'Encounter-open'
+            ? FhircastEncounterOpenContext
+            : EventName extends 'Encounter-close'
+              ? FhircastEncounterCloseContext
+              : EventName extends 'DiagnosticReport-open'
+                ? FhircastDiagnosticReportOpenContext
+                : EventName extends 'DiagnosticReport-close'
+                  ? FhircastDiagnosticReportCloseContext
+                  : EventName extends 'DiagnosticReport-update'
+                    ? FhircastDiagnosticReportUpdateContext
+                    : EventName extends 'DiagnosticReport-select'
+                      ? FhircastDiagnosticReportSelectContext
+                      : EventName extends 'syncerror'
+                        ? FhircastSyncErrorContext
+                        : never;
+
+export type FhircastPatientEvents = 'Patient-open' | 'Patient-close';
+export type FhircastImagingStudyEvents = 'ImagingStudy-open' | 'ImagingStudy-close';
+export type FhircastEncounterEvents = 'Encounter-open' | 'Encounter-close';
+export type FhircastDiagnosticReportEvents =
+  | 'DiagnosticReport-open'
+  | 'DiagnosticReport-close'
+  | 'DiagnosticReport-update'
+  | 'DiagnosticReport-select';
+export type FhircastAnchorResourceEvents =
+  | FhircastPatientEvents
+  | FhircastImagingStudyEvents
+  | FhircastEncounterEvents
+  | FhircastDiagnosticReportEvents;
+
+export type FhircastExtractAnchorResource<EventName extends FhircastAnchorResourceEvents> =
+  EventName extends FhircastPatientEvents
+    ? 'Patient'
+    : EventName extends FhircastImagingStudyEvents
+      ? 'ImagingStudy'
+      : EventName extends FhircastEncounterEvents
+        ? 'Encounter'
+        : EventName extends FhircastDiagnosticReportEvents
+          ? 'DiagnosticReport'
+          : never;
+
+export type CurrentContext<EventName extends FhircastAnchorResourceEvents | '' = FhircastAnchorResourceEvents | ''> =
+  EventName extends ''
+    ? { 'context.type': ''; context: never[] }
+    : EventName extends FhircastDiagnosticReportEvents
+      ? {
+          'context.type': 'DiagnosticReport';
+          'context.versionId': string;
+          context: (FhircastEventContext<'DiagnosticReport-open'> | FhircastHubContentContext)[];
+        }
+      : EventName extends FhircastPatientEvents | FhircastEncounterEvents | FhircastImagingStudyEvents
+        ? {
+            'context.type': FhircastExtractAnchorResource<EventName>;
+            'context.versionId': string;
+            context: FhircastEventContext<'Patient-open' | 'Encounter-open' | 'ImagingStudy-open'>;
+          }
+        : never;
+
 export type PendingSubscriptionRequest = Omit<SubscriptionRequest, 'endpoint'>;
 
-export type FhircastEventContextMap<EventName extends FhircastEventName = FhircastEventName> =
```

Comment:
Removed a lot of these monstrous types in favor of more explicit but more readable types